index.html

```diff
@@ -146,125 +149,353 @@ <h2>
+                equivalent to the current [=pointer-lock options=], process the
+                lock request and handle response upon fulfillment:
+                  <ol>
+                    <li>If the lock request failed:
+                      <ol>
+                        <li>[=Queue an element task=] on the [=user interaction
+                        task source=], given [=this=], to perform the following
+                        steps:
+                          <ol>
+                            <li>[=Fire an event=] named {{pointerlockerror}} at
+                            [=this=]'s [=Node/node document=].
+                            </li>
+                            <li>[=Reject=] <var>promise</var> with a
+                            "{{NotSupportedError}}" {{DOMException}}.
+                            </li>
+                          </ol>
+                        </li>
+                      </ol>
+                      <aside class="issue" data-number="91"></aside>
+                    </li>
+                    <li>If the lock request succeed:
+                      <ol>
+                        <li>Enter [=pointer lock state=] by performing the
+                        following:
+                          <ol>
+                            <li>Set the user agent's [=pointer-lock target=] to
+                            [=this=].
+                            </li>
+                            <li>Set the user agent's [=cursor position=] to the
+                            current system mouse location.
+                            </li>
+                            <li>Set the [=pointer-lock options=] to options.
+                            </li>
+                          </ol>
+                        </li>
+                        <li>[=Queue an element task=] on the [=user interaction
+                        task source=], given [=this=], to perform the following
+                        steps:
+                          <ol>
+                            <li>[=Fire an event=] named {{pointerlockchange}}
+                            at [=this=]'s [=Node/node document=].
+                            </li>
+                            <li>[=Resolve=] the <var>promise</var>.
+                            </li>
+                          </ol>
+                        </li>
+                      </ol>
+                    </li>
+                  </ol>
+                </li>
+              </ol>
+            </li>
+            <li>Return <var>promise</var>.
+            </li>
+          </ol>
           <aside class="note">
-            When a single <a>user activation</a> initiates both pointer lock
-            and fullscreen [[FULLSCREEN]], the {{requestPointerLock()}} call
-            succeeds only when it is made before a fullscreen request because
-            fullscreen is a <a>transient activation-consuming API</a>.
+            This algorithm is under active development and may be modified in
```

Comment:
I think we still need to mention how the entries in `lock requests queue` are handled, otherwise the returned `promise` will never be resolved, right?  Looks like we _need_ a Step 8 that posts a task to dequeue all entries.  Without an explicit text, we will leave open the incorrect (?) assumption that all entries will be synchronously dequeued at the end.

I am fine if you quote a new issue saying the `lock requests queue` processing is incomplete.

Reply:
Ack, will add new issue to my next commit.

Reply:
Issue #96 is created and associated in step 7 of the algorithm.